### PR TITLE
test(nodetool-stop): test nodetool stop for ICS

### PR DESCRIPTION
**Pull request by soyacz**
Testing nodetool stop when using ICS in Scylla Enterprise.

test job: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/lukasz/job/features-nodetool-stop-compaction-ics-test/
trello: https://trello.com/c/fMrUUoxc
## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by yarongilor** [comment]
consider changin order of lines in the duplicated method that fails the pre-commit pylint.

**Review by yarongilor — COMMENTED** [review]
Does new nodetool `stop` command applies only to ICS? or to all compaction strategies? is it tested for all compaction strategies?

File: defaults/aws_config.yaml [hunk 1]
```diff
@@ -8,22 +8,22 @@ instance_type_loader: 'c5.xlarge'
 instance_type_monitor: 't3.large'
 regions_data:
   us-east-1: # US East (N. Virginia)
-    ami_id_loader: 'ami-04d4eb45311c3d153' # Loader dedicated AMI v17
+    ami_id_loader: 'ami-04ba971b593b243c5' # Loader dedicated AMI v18
```
Comment (yarongilor) [inline defaults/aws_config.yaml:11]: perhaps better to consider adding this change in a separate PR since doesn't look related to new nodetool command enhancement.

File: stop_compaction_test.py [hunk 2]
```diff
@@ -247,3 +247,31 @@ def _grep_log_and_assert(self, node: BaseNode):
...
+        self.verify_stress_thread(cs_thread_pool=prepare)
+
+    def test_stop_major_compaction(self):
+        """Verifying data loss on stopping compaction by reading inserted"""
+        self.wait_no_compactions_running()
+        compaction_nemesis = StartStopMajorCompaction(
+            tester_obj=self,
+            termination_event=self.db_cluster.nemesis_termination_event)
+        compaction_nemesis.disrupt()
+        verify_cmd = "cassandra-stress read  cl=QUORUM -schema 'replication(factor=1) " \
```
Comment (yarongilor) [inline stop_compaction_test.py:272]: Not clear why using 'quorum' + rf=2. is it a test for only a single node? is there a need for 3 nodes?

**Comment by fgelcer** [comment]
ping @KnifeyMoloko @enaydanov

**Comment by fgelcer** [comment]
@soyacz , please look at the precommit failures

File: test-cases/features/stop-compaction-ics.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+n_monitor_nodes: 1
+
+instance_type_db: 'i3.large'
+
+backtrace_decoding: true
+
+user_prefix: 'stop-compaction-ics'
+append_scylla_yaml: |
+  enable_sstables_mc_format: true
+  enable_sstables_md_format: true
```
Comment (KnifeyMoloko) [inline test-cases/features/stop-compaction-ics.yaml:14]: This is only relevant for running sstables upgrade, I think we can safely remove it here.

**Review by KnifeyMoloko — CHANGES_REQUESTED** [review]
One simple code change request, otherwise looks ok. However I see the Jenkins tests are failing - is this expected? Maybe we'll need a filter if the current errors shouldn't invalidate the test.

File: defaults/aws_config.yaml (hunk 1 above)
Comment (soyacz) [reply defaults/aws_config.yaml:11]: yes, I'm waiting for https://github.com/scylladb/scylla-cluster-tests/pull/4587 to be merged. This one is needed for testing.

**Review by fgelcer — APPROVED** [review]
LGTM

**Review by yarongilor — COMMENTED** [review]
Shouldn't new nodetool command be tested for all compaction strategies?

**Comment by soyacz** [comment]
ping @KnifeyMoloko

**Review by KnifeyMoloko — DISMISSED** [review]

**Comment by fgelcer** [comment]
ping @fruch @enaydanov

**Comment by fgelcer** [comment]
ping @fruch @enaydanov @bentsi @roydahan

**Comment by fruch** [comment]
@soyacz @fgelcer please put relevant backport tags

**Comment by soyacz** [comment]
> @soyacz @fgelcer please put relevant backport tags

 https://github.com/scylladb/scylla-cluster-tests/pull/3924 was not backported anywhere (which this code bases on), I assume we don't do backports for this test. @fgelcer what do you think?

File: defaults/aws_config.yaml (hunk 1 above)
Comment (soyacz) [reply defaults/aws_config.yaml]: rebased

**Comment by fruch** [comment]
> > @soyacz @fgelcer please put relevant backport tags
> 
> #3924 was not backported anywhere (which this code bases on), I assume we don't do backports for this test. @fgelcer what do you think?

FYI #3924 code is in branch-5.0 and branch-2022.1, it was merged months ago

**Comment by soyacz** [comment]
> > > @soyacz @fgelcer please put relevant backport tags
> > 
> > 
> > #3924 was not backported anywhere (which this code bases on), I assume we don't do backports for this test. @fgelcer what do you think?
> 
> FYI #3924 code is in branch-5.0 and branch-2022.1, it was merged months ago

true, so this one can be backported too.

**Review by fgelcer — APPROVED** [review]
LGTM
